vavr/src/test/java/io/vavr/collection/AbstractTraversableTest.java

```diff
@@ -1157,6 +1189,31 @@ public void shouldThrowNPEWhenMaxOfIntAndNull() {
         of(1, null).max();
     }
 
+    @Test
+    public void shouldCalculateMaxOfDoublesContainingNaN() {
+        assertThat(of(1.0, Double.NaN, 2.0).max().get()).isEqualTo(Double.NaN);
+    }
+
+    @Test
+    public void shouldCalculateMaxOfFloatsContainingNaN() {
+        assertThat(of(1.0f, Float.NaN, 2.0f).max().get()).isEqualTo(Float.NaN);
+    }
+
+    @Test
+    public void shouldThrowClassCastExceptionWhenTryingToCalculateMaxOfDoubleAndFloat() {
+        assertThatThrownBy(() -> this.<Number> of(1.0, 1.0f).max()).isInstanceOf(ClassCastException.class);
```

Comment:
In previous tests you use `@Test(expected = ClassCastException.class)` but here you don't.

Reply:
Mmhh, interesting - will check that!

Reply:
Yes, `@Test(expected=...)` is the 'old style' we used. Assertions is now the way to go, we have more control over the check, e.g. instanceOf, hasCause, withMessage etc.

However, I plan to give all tests some love. We will migrate to JUnit 5.0. We will have more internal interfaces/mixins (prefixed with a `_`), like _Foldable, _NumericOps, ... These will have separate unit tests, which are also interfaces (that's possible with JUnit5).

This will us give better structure/control over our tests.